### PR TITLE
Update Spring Boot to 3.0.4

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind improvement
/area core

#### What this PR does / why we need it:

See https://github.com/halo-dev/halo/issues/3450 for more.

#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/3450

#### Does this PR introduce a user-facing change?

```release-note
升级 Spring Boot 至 3.0.4
```

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
```
Comment (ruibaby) [inline gradle/wrapper/gradle-wrapper.properties:3]: Gradle Wrapper 需要更新吗？

Comment (JohnNiang) [reply gradle/wrapper/gradle-wrapper.properties:3]: 已更新 gradle-wrapper.jar。

**Review by ruibaby — APPROVED** [review]
/approve

**Review by guqing — COMMENTED** [review]
/lgtm

**Pull request merged**